src/modules/sdexec/sdexec.c

```diff
@@ -0,0 +1,622 @@
+                                               void *value)
+{
+    const flux_msg_t *m;
+
+    m = flux_msglist_first (msglist);
+    while (m) {
+        if (flux_msg_aux_get (m, name) == value)
+            return m;
+        m = flux_msglist_next (msglist);
+    }
+    return NULL;
+}
+
+static void start_properties_changed (struct sdexec_ctx *ctx,
+                                      const flux_msg_t *msg,
+                                      json_t *properties)
+{
+    struct sdproc *proc = flux_msg_aux_get (msg, "sdproc");
+
+    json_int_t status = -1;
+    json_int_t code = -1;
+    json_int_t pid = -1;
+    const char *type = NULL;
+    const char *state = "";
+
+    (void)json_unpack (properties,
+                       "{s?[sI] s?[sI] s?[sI] s?[ss]}",
+                       "ExecMainStatus",    &type, &status,
+                       "ExecMainCode",      &type, &code,
+                       "ExecMainPID",       &type, &pid,
+                       "ActiveState",       &type, &state);
+#if 0
+    flux_log (ctx->h, LOG_DEBUG, "state=%s status=%jd code=%jd, pid=%jd",
+              state, (intmax_t)status, (intmax_t)code, (intmax_t)pid);
+#endif
+    if (streq (state, "failed"))
+        proc->failed = true;
+    else if (streq (state, "active"))
+        proc->activated = true;
+
+    if (pid > 0)
+        proc->pid = pid;
+
+    if (code > 0) {
+        proc->exit_code = code;
+        proc->exit_status = status;
+    }
+
+    if (proc->state == SD_INIT && proc->activated && proc->pid > 0) {
+        if (flux_respond_pack (ctx->h,
+                               msg,
+                               "{s:s s:I}",
+                               "type", "started",
+                               "pid", pid) < 0)
+            flux_log_error (ctx->h, "error responding to exec request");
+        proc->state = SD_RUNNING;
+    }
+    /* Exec failure etc - process was not started.  Assumes Type=exec.
+     * See list of process start-up types in Options section of:
+     * https://www.freedesktop.org/software/systemd/man/systemd.service.html
+     */
+    else if (proc->state == SD_INIT && proc->failed && proc->exit_code > 0) {
+        proc->state = SD_FINISHED;
+    }
+    else if (proc->state == SD_RUNNING
+        && proc->activated
+        && proc->exit_code > 0) {
+        int wait_status;
+
+        if (proc->exit_code == CLD_KILLED)
+            wait_status = __W_EXITCODE (0, proc->exit_status);
+        else
+            wait_status = __W_EXITCODE (proc->exit_status, 0);
+        if (flux_respond_pack (ctx->h,
+                               msg,
+                               "{s:s s:I}",
+                               "type", "finished",
+                               "status", wait_status) < 0)
+            flux_log_error (ctx->h, "error responding to exec request");
+        channel_start_output (proc->out);
+        channel_start_output (proc->err);
+        proc->state = SD_FINISHED;
+    }
```

Comment:
That looks wrong.  I guess it would mean all the output gets processed late, and most likely things will hang if there is significant output before that a kernel buffer fills up.  Not sure how that ended up like that - will  investigate/fix.  Thanks for catching it!